Tasks/ConsoleApp/AsyncAwait/ProductsRepository.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading.Tasks;
+using System.Linq;
+using Microsoft.EntityFrameworkCore;
+
+namespace AsyncAwait
+{
+    public interface IRepository<T>
+    {
+        Task<T> GetById(string id);
+        Task Save(T item);
+        Task Delete(string id);
+    }
+
+    public class MockProductsRepository : IRepository<Product>
+    {
+        private readonly ConcurrentDictionary<string, Product> _storage = new();
+
+        public Task Delete(string id)
+        {
+            _storage.TryRemove(id, out _);
+            return Task.CompletedTask;
+        }
+
+        public Task<Product> GetById(string id)
+        {
+            _storage.TryGetValue(id, out var value);
+            return Task.FromResult(value);
+        }
+
+        public Task Save(Product item)
+        {
+            _storage.AddOrUpdate(item.Id, item, (key, value) => item);
+            return Task.CompletedTask;
+        }
+    }
+
+    // Task. Implement repository to use real database MsSql почитать  with Entity Framework почитать (Code First) почитать 
+
+    public class ProductRepository : IRepository<Product>, IDisposable
+    {
+        public ProductsContext _context { get; set; }
```

Comment:
Wrong convention again. See below comment. And why this is public at all? You are injecting it and it means that you expect that injected implementation will be exactly what you injected. It strange that other classes can change it at any time.